src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,270 @@
+/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeApi = Interop.Diagnostics;
+
+namespace Tizen.Diagnostics
+{
+    /// <summary>
+    /// Enumeration for the context types.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public enum CtxType
+    {
+        /// <summary>
+        /// Crash context generated by crash-worker.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        Crash = 0,
+    }
+
+    /// <summary>
+    /// Enumeration for the report types.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public enum ReportType
+    {
+        /// <summary>
+        /// Full crash report (.tar.gz).
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        CrashFull = 0,
+        /// <summary>
+        /// Crash report in JSON format.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        CrashInfoJson,
+    }
+
+    /// <summary>
+    /// The Diagnostics class is used for requesting client data and receiving diagnostics report events.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public class Diagnostics
+    {
+        /// <summary>
+        /// Event to be invoked when new report context arrives.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public event EventHandler<ReportArrivedEventArgs> ReportArrived;
+
+        /// <summary>
+        /// The Diagnostics class constructor.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen> 
+        public Diagnostics()
+        {
+            this.SetNotificationCallback();
+        }
+
+        /// <summary>
+        /// Requests other apps to dump diagnostics data.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <feature>
+        /// http://tizen.org/privilege/internal/default/platform
+        /// </feature>
+        public DiagnosticsData RequestClientData(string clientId, string[] parameters)
```

Comment:
Could you provide more user-friendly API? User does not have any information about parameters to use this method.

Reply:
The problem is that the parameters are just passed as an array to the client and it depends on the client how it interprets it. It's just like calling an application with some argv. The Diagnostics API does not know anything about the parameters.